### ISSUE TITLE
No ads are collected

### ISSUE THREAD

**Issue by Demon-tk** [dhowe/AdNauseam#1778]
### Describe the issue
Ads are being blocked, as seen in uBlock Origin: [https://i.imgur.com/hjcPp3g.png](url)
Ads are not being collected or clicked: [https://i.imgur.com/lUGtfLj.png](url)

### Steps for anyone to reproduce the issue
Browse the web, have visited Youtube, Github, and Reddit (among others).

### Your settings

- OS/version: Windows 10 Insider Preview 10.0.21332.1010 
- Browser/version: Edge (chromium variant)
- AdNauseam version: AdNauseam v3.9.201
- Other extensions you have installed: Bitwarden Password Manager
- AdNauseam settings: Default settings w/ advanced user mode selected. See here: [https://i.imgur.com/kvJBcMf.png](url)

##### Your filter lists
Default filter lists + EasyList Cookie

### Your location/country
USA

**Comment by dhowe** [comment]
Two things to try:
1. Check the test page [here](https://rednoise.org/adntest/simple.html). You should see the add hidden/collected as below
2. Create a new firefox-profile as described [here](https://support.mozilla.org/en-US/kb/profile-manager-create-remove-switch-firefox-profiles#w_creating-a-profile) (a corrupt profile can often cause this behavior)

<img width="788" alt="image" src="https://user-images.githubusercontent.com/737638/112174467-66bed280-8c31-11eb-8ab4-0e55d9094e58.png">

**Comment by mage1k99** [comment]
hey @dhowe  is there any upcoming release ?

**Comment by dhowe** [comment]
yes, a big one -- should be within the next week, fingers crossed...

**Comment by Demon-tk** [comment]
@dhowe Upon visiting that page I did see that an ad was block, collected and clicked. 

I do not use Firefox, I am using edge.

**Comment by dhowe** [comment]
Ok, so basic functions are working - what page are you not seeing an ad collected (note that not ALL ads can be collected) ?

Perhaps try something like: https://edition.cnn.com/

**Comment by Demon-tk** [comment]
Taking a look at the page you [linked ](https://edition.cnn.com/), I am comparing uBlock origin on Vivaldi to AdNauseam on edge. The webpage looks the same, all the same, ads are hidden from view. However, the ads are not being collected by AdNauseam. 

This screenshot [here ](https://i.imgur.com/rSSPqUZ.png)shows the advanced panel for AdNauseam on Edge.

This t[hird screensho](https://i.imgur.com/WsytqXZ.png)t shows that there are no ads that are being clicked. 

Would you like to examine logs or any source of data on my end? I'm still not sure why this is happening. 

-Nate

**Comment by Demon-tk** [comment]
Is there more information I can provide that might help?

**Comment by dhowe** [comment]
@Demon-tk if you don't mind going to settings and clicking 'activate debugging mode', then visiting the page mentioned above with your 'browser console' open, then including the log output below, that would be quite helpful. 

<img width="445" alt="image" src="https://user-images.githubusercontent.com/737638/113035461-d7339980-91c5-11eb-81a7-422b5b6fe5f3.png">

Either Edge or Chromium should be OK, though Chromium has been tested more extensively...

**Comment by Demon-tk** [comment]
Here is my log file for [edition.cnn.com](https://gist.github.com/Demon-tk/29baa79a9692fc2fa4c80c7337551b0d)

**Comment by dhowe** [comment]
Thanks, I _think_ Edge's Tracking Prevention is conflicting with AdNauseam here. Can you include a screenshot of your Tracking Prevention settings from Edge preferences? You might also try disabling it, either globally or for that specific site, and retrying the page...

**Comment by dhowe** [comment]
Can you disable "Tracking Prevention" in the browser settings, then visit the following site and let me know whether ads are collected: https://www.scientificamerican.com/article/do-we-live-in-a-simulation-chances-are-about-50-50/
It should look as below:

<img width="399" alt="Screenshot 2021-04-02 at 3 01 20 AM" src="https://user-images.githubusercontent.com/737638/113341606-ed755d00-935f-11eb-847a-3878ca4fc4c4.png">

**Comment by Demon-tk** [comment]
Hey,

I've disabled tracking protection in Edge. Upon visiting the site you sent I get 0 ads blocked. You can see the tracking panel [here](https://i.imgur.com/zVof6TL.png).

I have also attached the log to a Gist [here](https://gist.github.com/Demon-tk/6902ab95a2af83a1ef2ed9c32e418142)

**Comment by dhowe** [comment]
Great, thanks for that info. @mneunomne I'm including the relevant part of the log below:

PAGE: https://www.scientificamerican.com/article/do-we-live-in-a-simulation-chances-are-about-50-50/ 

> 
> (index):76 GET https://static.chartbeat.com/js/chartbeat_mab.js net::ERR_BLOCKED_BY_CLIENT
> parser.js:302 [PARSER] Process(DIV) <div id=​"div-gpt-ad-interstitial-1" class=​"banner-interstitial">​…​</div>​
> parser.js:302 [PARSER] Checking children of <div id=​"div-gpt-ad-interstitial-1" class=​"banner-interstitial">​…​</div>​
> parser.js:302 [PARSER] No img found, check other cases <div id=​"div-gpt-ad-interstitial-1" class=​"banner-interstitial">​…​</div>​
> parser.js:302 [PARSER] findBgImage <div id=​"div-gpt-ad-interstitial-1" class=​"banner-interstitial">​…​</div>​
> parser.js:302 [PARSER] No images in children of <div id=​"div-gpt-ad-interstitial-1" class=​"banner-interstitial">​…​</div>​
> parser.js:302 [PARSER] Process(DIV) <div id=​"div-gpt-ad-interstitial-1-child" data-dfp-adcontainer=​"div-gpt-ad-interstitial-1">​</div>​
> parser.js:302 [PARSER] Checking children of <div id=​"div-gpt-ad-interstitial-1-child" data-dfp-adcontainer=​"div-gpt-ad-interstitial-1">​</div>​
> parser.js:302 [PARSER] No img found, check other cases <div id=​"div-gpt-ad-interstitial-1-child" data-dfp-adcontainer=​"div-gpt-ad-interstitial-1">​</div>​
> parser.js:302 [PARSER] findBgImage <div id=​"div-gpt-ad-interstitial-1-child" data-dfp-adcontainer=​"div-gpt-ad-interstitial-1">​</div>​
> parser.js:302 [PARSER] No images in children of <div id=​"div-gpt-ad-interstitial-1-child" data-dfp-adcontainer=​"div-gpt-ad-interstitial-1">​</div>​
> parser.js:302 [PARSER] Process(DIV) <div class=​"banner banner-leaderboard banner-leaderboard-top" data-banner-leaderboard>​…​</div>​
> parser.js:302 [PARSER] Checking children of <div class=​"banner banner-leaderboard banner-leaderboard-top" data-banner-leaderboard>​…​</div>​
> parser.js:302 [PARSER] No img found, check other cases <div class=​"banner banner-leaderboard banner-leaderboard-top" data-banner-leaderboard>​…​</div>​
> parser.js:302 [PARSER] findBgImage <div class=​"banner banner-leaderboard banner-leaderboard-top" data-banner-leaderboard>​…​</div>​
> parser.js:302 [PARSER] No images in children of <div class=​"banner banner-leaderboard banner-leaderboard-top" data-banner-leaderboard>​…​</div>​
> parser.js:302 [PARSER] Process(DIV) <div id=​"div-gpt-ad-top-1">​…​</div>​
> parser.js:302 [PARSER] Checking children of <div id=​"div-gpt-ad-top-1">​…​</div>​
> parser.js:302 [PARSER] No img found, check other cases <div id=​"div-gpt-ad-top-1">​…​</div>​
> parser.js:302 [PARSER] findBgImage <div id=​"div-gpt-ad-top-1">​…​</div>​
> parser.js:302 [PARSER] No images in children of <div id=​"div-gpt-ad-top-1">​…​</div>​
> parser.js:302 [PARSER] Process(DIV) <div id=​"div-gpt-ad-top-1-child" data-dfp-adcontainer=​"div-gpt-ad-top-1">​</div>​
> parser.js:302 [PARSER] Checking children of <div id=​"div-gpt-ad-top-1-child" data-dfp-adcontainer=​"div-gpt-ad-top-1">​</div>​
> parser.js:302 [PARSER] No img found, check other cases <div id=​"div-gpt-ad-top-1-child" data-dfp-adcontainer=​"div-gpt-ad-top-1">​</div>​
> parser.js:302 [PARSER] findBgImage <div id=​"div-gpt-ad-top-1-child" data-dfp-adcontainer=​"div-gpt-ad-top-1">​</div>​
> parser.js:302 [PARSER] No images in children of <div id=​"div-gpt-ad-top-1-child" data-dfp-adcontainer=​"div-gpt-ad-top-1">​</div>​
> parser.js:302 [PARSER] Process(FIGURE) <figure class=​"aside-banner small-hide medium-hide square-ad">​…​</figure>​
> parser.js:302 [PARSER] Checking children of <figure class=​"aside-banner small-hide medium-hide square-ad">​…​</figure>​
> parser.js:302 [PARSER] No img found, check other cases <figure class=​"aside-banner small-hide medium-hide square-ad">​…​</figure>​
> parser.js:302 [PARSER] findBgImage <figure class=​"aside-banner small-hide medium-hide square-ad">​…​</figure>​
> parser.js:302 [PARSER] No images in children of <figure class=​"aside-banner small-hide medium-hide square-ad">​…​</figure>​
> parser.js:302 [PARSER] Process(SECTION) <section class=​"footer-banner small-hide" data-footer-banner>​…​</section>​
> parser.js:302 [PARSER] Checking children of <section class=​"footer-banner small-hide" data-footer-banner>​…​</section>​
> parser.js:302 [PARSER] No img found, check other cases <section class=​"footer-banner small-hide" data-footer-banner>​…​</section>​
> parser.js:302 [PARSER] findBgImage <section class=​"footer-banner small-hide" data-footer-banner>​…​</section>​
> parser.js:302 [PARSER] No images in children of <section class=​"footer-banner small-hide" data-footer-banner>​…​</section>​
> (index):76 GET https://static.chartbeat.com/js/chartbeat.js net::ERR_BLOCKED_BY_CLIENT

**Comment by dhowe** [comment]
I've just re-tested with the latest edge and AdNauseam from the Edge store, using a US VPN, and I get the following for that page:

![image](https://user-images.githubusercontent.com/737638/113474558-38e45400-94a3-11eb-9d74-0fe202732189.png)

**Comment by Demon-tk** [comment]
I have tried 2 Windows 10 devices one on KB5000842 and the other on the latest insider build 19042.906.

I have not tried on Linux or MacOS.

**Comment by CorruptComputer** [comment]
Tried this on Firefox, both with Manjaro Linux and Windows 10 20H2. 
The ads are blocked, but AdNauseam doesn't seem to be counting them.
The test page you had sent (https://rednoise.org/adntest/simple.html) does show an ad blocked and clicked, but the other site suggested (https://edition.cnn.com/) does not.

**Comment by Demon-tk** [comment]
Yes @CorruptComputer I find the exact same thing. The only difference with my setup is that I am on Windows 10 prerelease build 21354.1.

**Comment by dhowe** [comment]
@mneunomne can you do some further testing here?

**Comment by mneunomne** [comment]
I do not have a windows machine to test this on at the moment. But it seems that Ad collection is working fine in Edge v89.0.774.76 in both simple tests cases and "real-case" ones like [this page](https://www.scientificamerican.com/article/dark-matters-last-stand/).
I will try to get access to a windows machine to do some further tests.

**Comment by dhowe** [comment]
@Real-John-Cheung maybe you can help with this?

**Comment by Demon-tk** [comment]
@mneunomne You can always spin up a VM real quick, MS hosts developer pre-built images for free.

**Comment by Real-John-Cheung** [comment]
On Windows 10 Enterprise 1909 (the version thing is a bit confusing for me... this is the value of the 'version' field in setting)
with the same setting as shown in [this image](https://i.imgur.com/kvJBcMf.png) and Bitwarden Password Manager installed, tracking prevention disabled.
AdNauseam v3.9.201

In short: the described problem occurs on Edge/Chrome when visit edition.cnn.com; but it doesn't seem to occur on Firefox, also not in the Scientific American page... maybe the problem is Chromium-related and is triggered by specific kinds of ads

note: seems that text ads from CNN are captured, maybe we can narrow it down to the image ads?
(see the screenshots on Edge below)
<hr>


### Firefox 87.0 (newest official version on 16/4/21)
Seems in both sites mentioned above
 it works well
<b>CNN</b>
![Screenshot (2)](https://user-images.githubusercontent.com/61215358/115027240-0c462880-9ef6-11eb-8daf-6261c87de8f6.png)

<b>Scientific American</b>
![Screenshot (1)](https://user-images.githubusercontent.com/61215358/115027320-241dac80-9ef6-11eb-89f5-ec66e58c3e30.png)
![Screenshot (3)](https://user-images.githubusercontent.com/61215358/115028533-8f1bb300-9ef7-11eb-9457-26c1e5c4971a.png)
<hr>


### Chrome 90.0.4430.72 (newest official build on 16/4/21)
<b>CNN</b>: ads are blocked but not captured by adNauseam
![Screenshot (8)](https://user-images.githubusercontent.com/61215358/115031594-f0915100-9efa-11eb-8d74-da459cff4b72.png)
![Screenshot (9)](https://user-images.githubusercontent.com/61215358/115031621-f71fc880-9efa-11eb-825b-098024952372.png)
logs from the AdNauseam logger: 
[logs_cnn_chrome.md](https://github.com/dhowe/AdNauseam/files/6325372/logs_cnn_chrome.md)

<b>Scientific American</b>
works well
![Screenshot (6)](https://user-images.githubusercontent.com/61215358/115031878-4108ae80-9efb-11eb-933e-4a162b9b8e89.png)
<hr>


### Edge 90.0.818.39 (newest official build on 16/4/21)
<b>CNN</b>: ads are blocked but not captured by adNauseam (but a text ad is captured, as shown in the third screenshot below, so maybe only image ad will trigger this)
![Screenshot (11)](https://user-images.githubusercontent.com/61215358/115035474-305a3780-9eff-11eb-80ac-37426612e461.png)
![Screenshot (12)](https://user-images.githubusercontent.com/61215358/115035445-2a645680-9eff-11eb-835e-b909e38ada24.png)
![Screenshot (15)](https://user-images.githubusercontent.com/61215358/115034785-7793f880-9efe-11eb-9f8e-a9599fff727b.png)
logs from the AdNauseam logger:
[logs_cnn_edge.md](https://github.com/dhowe/AdNauseam/files/6325591/logs_cnn_edge.md) 

<b>Scientific American</b>
works well
![Screenshot (14)](https://user-images.githubusercontent.com/61215358/115035616-554eaa80-9eff-11eb-93a0-349f8941ffc6.png)

**Comment by CorruptComputer** [comment]
Was this fixed? This is still happening for me.

**Comment by dhowe** [comment]
This has been fixed, but the new release is still in the Firefox review queue -- it should be released very shortly. thanks

**Comment by Demon-tk** [comment]
Hello all who have been helpful in resolving this issue. Upon updating to the newest release: [adnauseam-3.10.4.edge.zip](https://github.com/dhowe/AdNauseam/releases/tag/v3.10.4) I visited this [site ](https://www.scientificamerican.com/article/do-we-live-in-a-simulation-chances-are-about-50-50/)to test adblocking. 

I still see that there are no advertisements shown as clicked: 
![image](https://user-images.githubusercontent.com/8584243/117323462-e5c43f00-ae5c-11eb-9200-058a3a729149.png)

But uBlock Origin is functional 
![image](https://user-images.githubusercontent.com/8584243/117323527-f5438800-ae5c-11eb-8214-5e1d71bf61ac.png)

I do not believe the issue has been fully resolved.

**Comment by Tamnac** [comment]
I can confirm what @Demon-tk says. I have also updated to the latest version, and while ads are blocked, they are not collected. On vivaldi (chromium) in a fresh profile, no other extensions, built-in blocking disabled.

**Comment by dhowe** [comment]
Thanks. Please re-check the page [here](https://rednoise.org/adntest/simple.html) if you don't mind.

**Comment by Demon-tk** [comment]
When viewing that page the ad is not blocked or clicked: 
![image](https://user-images.githubusercontent.com/8584243/117472688-26888a80-af27-11eb-81ec-16db423507ff.png)

You can see that uBlock Origin also does not detect it as an ad: 
![image](https://user-images.githubusercontent.com/8584243/117472744-36a06a00-af27-11eb-937b-8d4f81710466.png)

**Comment by dhowe** [comment]
@Demon-tk We are interested more here in whether the ads are collected. My assumption is that collection (and clicking) works correctly on the rednoise test page, but not on cnn.com. Is this correct?

**Comment by Demon-tk** [comment]
@dhowe Yes, they are collected on Rednose but not on cnn.

**Comment by CorruptComputer** [comment]
I still have the same experience as @Demon-tk using Firefox 88.0.1 on Windows 10 Education 20H2 19042.985

**Comment by dhowe** [comment]
Please try this test page: [https://www.atptour.com/](https://www.atptour.com/). My result is below:

<img width="1057" alt="image" src="https://user-images.githubusercontent.com/737638/118346782-66c3ba80-b570-11eb-94dd-53e5d7208e9d.png">

**Comment by CorruptComputer** [comment]
@dhowe That site seems to be working fine for me:
![image](https://user-images.githubusercontent.com/5573038/118348219-d86f1a80-b516-11eb-8404-0a5dd722b168.png)

Its not on every site that this doesn't seem to work, I just notice that a large number of them seem to have this issue.

**Comment by dhowe** [comment]
This issue _should_ be resolved in the version 3.11, to be released shortly. If it persists, please re-open... (thanks)

**Comment by FOREVEREALIZE** [comment]
This is still happening for me. Using AdNauseam v3.18.0 on this [Scientific American Article](https://www.scientificamerican.com/article/do-we-live-in-a-simulation-chances-are-about-50-50/) the ads are blocked by uBlock, but not counted nor clicked by AdNauseam.

**Comment by FOREVEREALIZE** [comment]
@dhowe I believe this issue should be reopened.

I am using Linux Mint 21.1 Cinnamon with Firefox 11.0.2. AdNauseam version is mentioned above. I use PiHole but have also tried this on a clean VM not going through it, same result.

**Comment by dhowe** [comment]
Am I correct that ads are correctly hidden, but not collected/clicked on the [Scientific American Article](https://www.scientificamerican.com/article/do-we-live-in-a-simulation-chances-are-about-50-50/) ?

**Comment by mneunomne** [comment]
Yes, correct. We had once implemented collection in this page, but then now it's not working again. Maybe make a separate issue ticket for it.

**Comment by dhowe** [comment]
@FOREVEREALIZE is this (Scientific America) the only page/site on which you are seeing this issue?

**Comment by FOREVEREALIZE** [comment]
> @FOREVEREALIZE is this (Scientific America) the only page/site on which you are seeing this issue?

No, I've also tried [CNN Editor](https://editor.cnn.com) with the same result. YouTube and Fandom also do this.
